docs/source/extensions_tutorial/7_publishing_extensions.rst

```diff
@@ -1,4 +1,114 @@
-Publishing Extensions
+Publishing extensions
 ---------------------
 
-Neurodata extensions can be shared with the community using the `NDX Catalog <https://nwb-extensions.github.io/>`_
\ No newline at end of file
+.. image:: ../img/publishing_extensions.png
+   :width: 315px
+   :class: align-right
+
+Neurodata extensions can be shared with the community using the :ndx-catalog:`NDX Catalog <>`.
+As illustrated in the figure, the publication process is divided into three main steps:
+1) open release of the sources to the community using git, 2) open access of versioned
+releases via PyPI, and 3) open publication of the extension to the community via the
+:ndx-catalog:`NDX Catalog <>`.
+
+Releasing your extension on GitHub
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+1. Before publishing your licence online you should add a license file. Permissive licenses should be used if possible. A `BSD license <https://opensource.org/licenses/BSD-3-Clause>` is recommended.
+2. Modify ``README.md`` at the root directory of your extension repo to describe the extension for interested developers and users.
+3. The first step to publishing your extension then is to make your Git repository accessible online
+   via GitHub, or any other public Git hosting service of your choice. To publish your extension
+   on GitHub you will need a `GitHub account <https://github.com/>`_ and follow the following
+   `instructions to add an existing project to GitHub <https://help.github.com/en/articles/adding-an-existing-project-to-github-using-the-command-line>`
+4. Make a release for the extension on GitHub with the version number specified. e.g. if the version
+   is ``0.1.0``, then this page should exist: https://github.com/<my_username>/<my_extension>/releases/tag/0.1.0``.
+   See the `creating a release guide <https://help.github.com/en/github/administering-a-repository/creating-releases>` on GitHub for instructions on how to make a release on GitHub.
```

Comment:
recommend semantic versioning

Reply:
I added a note to link to versioning guidelines